### PR TITLE
[Add]: Relationship between StorageClass and other k8s components

### PR DESCRIPTION
**Pull request by pandeyshubham03**
**Notes for Reviewers**

This PR fixes #

1. Adds Relationship between StorageClass and other k8s components.

Working proof of this PR:
https://github.com/meshery/meshery/assets/128715606/c284c457-acc6-4434-9c89-1e5156fb5192

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@pandeyshubham03 would you like to offer review here?

**Comment by pandeyshubham03** [comment]
I currently have the read-only access to the [spreadsheet](https://docs.google.com/spreadsheets/d/1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdwizOJmeMw/edit#gid=1646054916). Kindly allow me to edit it in order to make necessary changes.

**Comment by RipulHandoo** [comment]
**_Note:_** [Spreadsheet ](https://docs.google.com/spreadsheets/d/1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdwizOJmeMw/edit#gid=1646054916)updated.  ✅

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -1077,6 +1077,40 @@
...
+        "from":[
+          {
+            "kind": "StorageClass",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "A StorageClass in Kubernetes defines the type and properties of storage that can be dynamically provisioned for persistent volumes."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1092]: Please add a description of how one component modifies or is modified by the other.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -1293,6 +1293,40 @@
...
+        "from":[
+          {
+            "kind": "StorageClass",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "A StorageClass in Kubernetes defines the type and properties of storage that can be dynamically provisioned for persistent volumes and its name is mentioned in a PersistentVolumeClaim template."
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1308]:
```suggestion
              "description": "StorageClass is used in conjunction with PVC that allow Pods to dynamically request a new storage."
```

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**